tcxparser/tcxparser.py

```diff
@@ -98,6 +98,39 @@ def hr_min(self):
         """Minimum heart rate of the workout"""
         return min(self.hr_values())
 
+    def hr_percent_in_zones(self, zones):
+        """Percentage of workout spent in eacht heart rate zone.
+
+        Given these user's heart rate zones:
+        {
+            "Z0": (0, 119),
+            "Z1": (120, 199),
+            "Z2": (200, 240),
+        }
+
+        We return something like:
+        {
+            "Z0": 5,
+            "Z1": 95,
+            "Z2": 0,
+        }
+
+        Correct calculation relies on evenly spaced measurement times.
+        """
+        # count number of HR measurements per zone
+        per_zone = dict(zip(zones.keys(), [0 for x in zones.keys()]))
```

Comment:
I now use Python's cool `defaultdict` for this. It's a dictionary that acts like a normal dictionary unless you try to assign values to an unknown key, in which case, it will create an entry for that key, defaulting to whatever you'd like it to default to. So this line could be replaced with:
```
from collections import defaultdict
per_zone = defaultdict(int)  # <- int class defaults to zero
```

Reply:
This code could indeed be more elegant. I found `defaultdict` by itself does not do the main thing that i want to happen here, which is create a dictionary with same keys as `zones`. I propose to change it to this:
```
        # Initialize a dictionary with one item for each zone
        per_zone = dict.fromkeys(zones.keys(), 0)
```

Reply:
Ahh good point. This looks great.